robolectric-shadows/shadows-core/src/main/java/org/robolectric/fakes/RoboCursor.java

```diff
@@ -60,6 +63,26 @@ public void close() {
     closeWasCalled = true;
   }
 
+  @Override
+  public int getCount() {
+    return results.length;
+  }
+
+  @Override
+  public int getColumnCount() {
+    return columnNames.size();
+  }
+
+  @Override
+  public String getColumnName(int index) {
+    return columnNames.get(index);
+  }
+
+  @Override
+  public int getType(int columnIndex) {
+    return DatabaseUtils.getTypeOfObject(results[0][columnIndex]);
```

Comment:
Are you referring to getType? No, because if you moved through the cursor and then ask for the type, it will blow up (resultsIndex will be out of bounds).